runner/engine_observability.py

```
@@ -570,6 +570,102 @@ def _format_uncommitted_for_log(state: dict[str, str]) -> str:
+        wd = str(workdir)
+        result = dict(empty)
+        if base_sha:
+            count = subprocess.run(
+                ["git", "-C", wd, "rev-list", "--count", f"{base_sha}..HEAD"],
+                capture_output=True, text=True, timeout=10, check=False,
+                stdin=subprocess.DEVNULL,
+            )
+            if count.returncode == 0 and count.stdout.strip().isdigit():
+                result["commits_created"] = count.stdout.strip()
+        if base_upstream_sha:
+            upstream = subprocess.run(
```

Ruff (0.16.3)

[error] `subprocess` call: check for execution of untrusted input

(S603)